spec/models/service_template_ansible_playbook_spec.rb

```diff
@@ -0,0 +1,38 @@
+describe ServiceTemplateAnsiblePlaybook do
+  describe '#create_catalog_item' do
+    let(:user) { FactoryGirl.create(:user_with_group) }
+    let(:ems) { FactoryGirl.create(:automation_manager_ansible_tower) }
+    let(:inventory_root_group) { FactoryGirl.create(:inventory_root_group, :ems_ref => 'many') }
+    let(:config_script) { FactoryGirl.create(:configuration_script) }
+    let(:script_source) { FactoryGirl.create(:configuration_script_source, :manager => ems) }
+    let(:playbook) { FactoryGirl.create(:configuration_script_payload,
+                                        :configuration_script_source => script_source,
+                                        :inventory_root_group => inventory_root_group,
+                                        :type => 'ManageIQ::Providers::AnsibleTower::AutomationManager::Playbook') }
+    let(:service_template_catalog) { FactoryGirl.create(:service_template_catalog) }
+    let(:catalog_item_options) do
+      {
+        :name         => 'test_ansible_catalog_item',
+        :description  => 'test ansible',
+        :service_template_catalog => service_template_catalog,
+        :config_info  => {
+          :provision               => {
+            :new_dialog_name => 'test_dialog',
+            :hosts => 'many',
+            :credential => 6,
+            :playbook_id => playbook.id,
+          },
+        }
+      }
+    end
+
+    it 'creates and returns a catalog item' do
+      expect(ManageIQ::Providers::AnsibleTower::AutomationManager::ConfigurationScript).to receive(:create_in_provider_queue).and_return(10)
+      expect(MiqTask).to receive(:wait_for_taskid).with(10).and_return(config_script)
```

Comment:
This can be re done - at the time I thought I would need the `.with()` call - but I believe `to receive(:wait_for_taskid)` should be enough.